database/ticket.go

```diff
@@ -114,23 +126,128 @@ func (vdb *VspDatabase) InsertNewTicket(ticket Ticket) error {
 		if err != nil {
-			return fmt.Errorf("could not marshal ticket: %w", err)
+			return fmt.Errorf("putting ticket in bucket failed: %w", err)
 		}
 
-		return ticketBkt.Put(hashBytes, ticketBytes)
+		return nil
 	})
 }
 
-func (vdb *VspDatabase) DeleteTicket(ticket Ticket) error {
+func putTicketInBucket(bkt *bolt.Bucket, ticket *Ticket) error {
+	var err error
+	if err = bkt.Put(HashK, []byte(ticket.Hash)); err != nil {
+		return err
+	}
+	if err = bkt.Put(CommitmentAddressK, []byte(ticket.CommitmentAddress)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeAddressK, []byte(ticket.FeeAddress)); err != nil {
+		return err
+	}
+	if err = bkt.Put(VotingWIFK, []byte(ticket.VotingWIF)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeTxHexK, []byte(ticket.FeeTxHex)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeTxHashK, []byte(ticket.FeeTxHash)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeTxStatusK, []byte(ticket.FeeTxStatus)); err != nil {
+		return err
+	}
+	if err = bkt.Put(OutcomeK, []byte(ticket.Outcome)); err != nil {
+		return err
+	}
+	if err = bkt.Put(PurchaseHeightK, int64ToBytes(ticket.PurchaseHeight)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeAddressIndexK, uint32ToBytes(ticket.FeeAddressIndex)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeAmountK, int64ToBytes(ticket.FeeAmount)); err != nil {
+		return err
+	}
+	if err = bkt.Put(FeeExpirationK, int64ToBytes(ticket.FeeExpiration)); err != nil {
+		return err
+	}
+
+	confirmed := []byte{0}
+	if ticket.Confirmed {
+		confirmed = []byte{1}
+	}
+	if err = bkt.Put(ConfirmedK, confirmed); err != nil {
+		return err
+	}
+
+	// Ensure a bucket for vote choices exists.
+	voteBkt, err := bkt.CreateBucketIfNotExists(VoteChoicesK)
+	if err != nil {
+		return err
+	}
+
+	// Remove any existing vote choices.
+	err = voteBkt.ForEach(func(k, v []byte) error {
+		return voteBkt.Delete(k)
+	})
+	if err != nil {
+		return err
+	}
+
+	// Insert vote choices.
+	for agenda, choice := range ticket.VoteChoices {
+		if err = voteBkt.Put([]byte(agenda), []byte(choice)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getTicketFromBkt(bkt *bolt.Bucket) (*Ticket, error) {
+	var ticket Ticket
+
+	ticket.Hash = string(bkt.Get(HashK))
+	ticket.CommitmentAddress = string(bkt.Get(CommitmentAddressK))
+	ticket.FeeAddress = string(bkt.Get(FeeAddressK))
+	ticket.VotingWIF = string(bkt.Get(VotingWIFK))
+	ticket.FeeTxHex = string(bkt.Get(FeeTxHexK))
+	ticket.FeeTxHash = string(bkt.Get(FeeTxHashK))
+	ticket.FeeTxStatus = FeeStatus(bkt.Get(FeeTxStatusK))
+	ticket.Outcome = TicketOutcome(bkt.Get(OutcomeK))
+
+	ticket.PurchaseHeight = bytesToInt64(bkt.Get(PurchaseHeightK))
+	ticket.FeeAddressIndex = bytesToUint32(bkt.Get(FeeAddressIndexK))
+	ticket.FeeAmount = bytesToInt64(bkt.Get(FeeAmountK))
+	ticket.FeeExpiration = bytesToInt64(bkt.Get(FeeExpirationK))
+
+	// TODO is this dodgy?
+	if bkt.Get(ConfirmedK)[0] == byte(1) {
+		ticket.Confirmed = true
+	}
```

Comment:
Highlighting this particularly icky piece of code for reviewers - any better ideas for how to encode/decode `bool` & `[]byte`?